pkg/scheduler/objects/application.go

```diff
@@ -2118,3 +2119,34 @@ func (sa *Application) updateRunnableStatus(runnableInQueue, runnableByUserLimit
 	}
 	sa.runnableByUserLimit = runnableByUserLimit
 }
+
+// GetGuaranteedResource returns the guaranteed resource that was set in the application tags
+func (sa *Application) GetGuaranteedResource() *resources.Resource {
+	return sa.getResourceFromTags(siCommon.AppTagNamespaceResourceGuaranteed)
+}
+
+// GetMaxResource returns the max resource that was set in the application tags
+func (sa *Application) GetMaxResource() *resources.Resource {
+	return sa.getResourceFromTags(siCommon.AppTagNamespaceResourceQuota)
+}
+
+func (sa *Application) getResourceFromTags(tag string) *resources.Resource {
+	value := sa.GetTag(tag)
+	if value == "" {
+		return nil
+	}
+
+	resource, err := resources.NewResourceFromString(value)
+	if err != nil {
+		log.Log(log.SchedQueue).Warn("application resource conversion failure",
+			zap.String("tag", tag),
+			zap.String("json string", value),
+			zap.Error(err))
+	} else if !resources.StrictlyGreaterThanZero(resource) {
+		log.Log(log.SchedQueue).Warn("resource quantities should be greater than zero",
+			zap.Stringer("maxResource", resource))
```

Comment:
`resource` could be guaranteed resource now.

Reply:
Fixed